apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/WelcomeQuestionForm.tsx

```diff
@@ -0,0 +1,204 @@
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(payload),
+      });
+
+      if (!response.ok) {
+        throw new Error(`Upload failed with status: ${response.status}`);
+      }
+
+      return response.json();
+    } catch (error) {
+      console.error("Upload error:", error);
+      throw error;
+    }
+  };
+
+  return (
+    <form>
+      <div className="mt-3">
+        <Label htmlFor="headline">Question</Label>
+        <div className="mt-2">
+          <Input
+            autoFocus
+            id="headline"
+            name="headline"
+            value={question.headline}
+            onChange={(e) => updateQuestion(questionIdx, { headline: e.target.value })}
+            isInvalid={isInValid && question.headline.trim() === ""}
+          />
+        </div>
+      </div>
+      <div className="mt-3">
+        <Label htmlFor="subheader">Description</Label>
+        <div className="mt-2">
+          <Editor
+            getText={() =>
+              md.render(
+                question.html || "We would love to talk to you and learn more about how you use our product."
+              )
+            }
+            setText={(value: string) => {
+              updateQuestion(questionIdx, { html: value });
+            }}
+            excludedToolbarItems={["blockType"]}
+            disableLists
+            firstRender={firstRender}
+            setFirstRender={setFirstRender}
+          />
+        </div>
+      </div>
+      <div className="mt-3 flex w-full items-center justify-center">
+        <label
+          htmlFor="companyLogo"
+          className="relative flex h-64 w-full cursor-pointer flex-col items-center justify-center rounded-lg border-2 border-dashed border-gray-300 bg-gray-50 hover:bg-gray-100 dark:border-gray-600 dark:bg-gray-700 dark:hover:border-gray-500 dark:hover:bg-gray-600 dark:hover:bg-gray-800"
+          onDragOver={(e) => handleDragOver(e)}
+          onDrop={(e) => handleDrop(e)}>
+          {question.companyLogo ? (
+            <>
+              <img
+                src={question.companyLogo}
+                alt="Company Logo"
+                className="max-h-full max-w-full rounded-lg object-contain"
+              />
+              <div className="absolute inset-0 flex items-center justify-center bg-black bg-opacity-20 transition-opacity duration-300 hover:bg-opacity-60">
+                <label htmlFor="companyLogo" className="cursor-pointer text-sm font-semibold text-white">
+                  Modify
+                </label>
+              </div>
+            </>
+          ) : (
+            <div className="flex flex-col items-center justify-center pb-6 pt-5">
+              <ArrowUpTrayIcon className="h-6 text-gray-500" />
+              <p className="mt-2 text-sm text-gray-500 dark:text-gray-400">
+                <span className="font-semibold">Click or drag to upload files.</span>
+              </p>
+            </div>
+          )}
+          <input
```

Comment:
This file input component is very well done! Please move this into the `@formbricks/ui` package in a seperate `FileInput` component 🙏 .

Keep the input element and the other jsx in the `FileInput/index.tsx` file and keep all the logic for handling uploading of files under `FileInput/lib` directory. This will make sure that the component and the underlying logic for calling the storage api for uploading files is reusable 🙂

Reply:
Done and tried my best to make it as reusable as possible : )